examples/cross-browser.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2021 Google LLC.
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script implemets Puppeteer's `examples/cross-browser.js` scenario using WebDriver BiDi.
+# https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js
+
+import asyncio
+import json
+import os
+import pytest
+import websockets
+import time
+
+async def get_websocket():
+    port = os.getenv('PORT', 8080)
+    url = f'ws://localhost:{port}'
+    return await websockets.connect(url)
+
+async def send_JSON_command(websocket, command):
+    await websocket.send(json.dumps(command))
+
+async def read_JSON_message(websocket):
+    return json.loads(await websocket.recv())
+
+async def main():
+    # const browser = await puppeteer.launch();
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L29
+    websocket = await get_websocket()
+
+    # Receive initial `browsingContext.contextCreated` event.
+    resp = await read_JSON_message(websocket)
+    assert resp['method'] == 'browsingContext.contextCreated'
```

Comment:
Leaving the asserts makes this look a lot like a test, and not like what a real client might do. What do you think about adding a simple helper that waits for the "browsingContext.contextCreated" event but doesn't fail if some other event comes first?

Similarly for commands, waiting for the command response and not failing if there's another message that comes first.

I guess what this has to be is one bit of code that reads messages and dispatches, and it probably has a list of `asyncio.Future` objects, calling `set_result` or `set_exception`.

Reply:
Agree. Removed assertions and extracted waiting for a command logic to a dedicated function.